### ISSUE TITLE
[NativeAOT-LLVM]: llvm clrjit assert on Release dll input in PHASE_MORPH_GLOBAL

### ISSUE THREAD

**Issue by yowl** [dotnet/runtimelab#1803]
For https://github.com/dotnet/runtimelab/blob/ce43f371e1761fe6335f8c63d344e605b892ebe9/src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/WellKnownTypeExtensions.cs#L13

The llvm clrjit is hitting the assert at https://github.com/dotnet/runtimelab/blob/ce43f371e1761fe6335f8c63d344e605b892ebe9/src/coreclr/jit/morph.cpp#L16131

when compiling the release IL build of the HelloWasm test (and no doubt all the tests).

It seems to be saying this must a ret void, but as you can see that's not going to be the case in this method, and indeed the `ret` is 
```
0x0000026b43cf5ea0 [GT_RETURN, TYP_STRUCT]
```
So presumably there's no return blocks to be merged in this method and we shouldn't be here.  

The ngendump output is attached for both the debug (works) and release( fails) inputs.

The stack is

```
 	KernelBase.dll!00007ffffad99072()	Unknown
 	clrjit_browser_wasm32_x64.dll!assertAbort(const char * why, const char * file, unsigned int line) Line 306	C++
 	clrjit_browser_wasm32_x64.dll!noWayAssertAbortHelper(const char * cond, const char * file, unsigned int line) Line 512	C++
 	clrjit_browser_wasm32_x64.dll!noWayAssertBodyConditional(const char * cond, const char * file, unsigned int line) Line 529	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::fgMergeBlockReturn(BasicBlock * block) Line 16131	C++
>	clrjit_browser_wasm32_x64.dll!Compiler::fgMorphBlocks() Line 16006	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::compCompile::__l2::<lambda>() Line 4959	C++
 	clrjit_browser_wasm32_x64.dll!ActionPhase<void <lambda>(void)>::DoPhase() Line 65	C++
 	clrjit_browser_wasm32_x64.dll!Phase::Run() Line 61	C++
 	clrjit_browser_wasm32_x64.dll!DoPhase<void <lambda>(void)>(Compiler * _compiler, Phases _phase, Compiler::compCompile::__l2::void <lambda>(void) _action) Line 79	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::compCompile(void * * methodCodePtr, unsigned int * methodCodeSize, JitFlags * compileFlags) Line 5016	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::compCompileHelper(CORINFO_MODULE_STRUCT_ * classPtr, ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, void * * methodCodePtr, unsigned int * methodCodeSize, JitFlags * compileFlags) Line 6556	C++
 	clrjit_browser_wasm32_x64.dll!`Compiler::compCompile'::`73'::__Body::Run(Compiler::compCompile::__l2::__JITParam * __JITpParam) Line 5825	C++
 	clrjit_browser_wasm32_x64.dll!Compiler::compCompile(CORINFO_MODULE_STRUCT_ * classPtr, void * * methodCodePtr, unsigned int * methodCodeSize, JitFlags * compileFlags) Line 5829	C++
 	clrjit_browser_wasm32_x64.dll!``jitNativeCode'::`8'::__Body::Run'::`6'::__Body::Run(jitNativeCode::__l8::__Body::Run::__l5::__JITParam * __JITpParam) Line 7199	C++
 	clrjit_browser_wasm32_x64.dll!`jitNativeCode'::`8'::__Body::Run(jitNativeCode::__l2::__JITParam * __JITpParam) Line 7202	C++
 	clrjit_browser_wasm32_x64.dll!jitNativeCode(CORINFO_METHOD_STRUCT_ * methodHnd, CORINFO_MODULE_STRUCT_ * classPtr, ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, void * * methodCodePtr, unsigned int * methodCodeSize, JitFlags * compileFlags, void * inlineInfoPtr) Line 7226	C++
 	clrjit_browser_wasm32_x64.dll!CILJit::compileMethod(ICorJitInfo * compHnd, CORINFO_METHOD_INFO * methodInfo, unsigned int flags, unsigned char * * entryAddress, unsigned int * nativeSizeOfCode) Line 279	C++
 	jitinterface_x64.dll!JitCompileMethod(CorInfoExceptionClass * * ppException, ICorJitCompiler * pJit, void * thisHandle, void * * callbacks, CORINFO_METHOD_INFO * methodInfo, unsigned int flags, unsigned char * * entryAddress, unsigned int * nativeSizeOfCode) Line 36	C++
 	[External Code]	
 	ILCompiler.RyuJit.dll!Internal.JitInterface.CorInfoImpl.CompileMethodInternal(ILCompiler.DependencyAnalysis.IMethodNode methodCodeNodeNeedingCode, Internal.IL.MethodIL methodIL) Line 305	C#
 	ILCompiler.RyuJit.dll!Internal.JitInterface.CorInfoImpl.CompileMethod(ILCompiler.DependencyAnalysis.IMethodCodeNode methodCodeNodeNeedingCode, Internal.IL.MethodIL methodIL) Line 62	C#
 	ILCompiler.LLVM.dll!ILCompiler.LLVMCodegenCompilation.CompileSingleMethod(Internal.JitInterface.CorInfoImpl corInfo, ILCompiler.DependencyAnalysis.LLVMMethodCodeNode methodCodeNodeNeedingCode) Line 143	C#
 	ILCompiler.LLVM.dll!ILCompiler.LLVMCodegenCompilation.CompileSingleThreaded(System.Collections.Generic.List<ILCompiler.DependencyAnalysis.LLVMMethodCodeNode> methodsToCompile) Line 121	C#
 	ILCompiler.LLVM.dll!ILCompiler.LLVMCodegenCompilation.ComputeDependencyNodeDependencies(System.Collections.Generic.List<ILCompiler.DependencyAnalysisFramework.DependencyNodeCore<ILCompiler.DependencyAnalysis.NodeFactory>> obj) Line 104	C#
 	ILCompiler.DependencyAnalysisFramework.dll!ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer<ILCompiler.DependencyAnalysisFramework.NoLogStrategy<ILCompiler.DependencyAnalysis.NodeFactory>, ILCompiler.DependencyAnalysis.NodeFactory>.ComputeDependencies(System.Collections.Generic.List<ILCompiler.DependencyAnalysisFramework.DependencyNodeCore<ILCompiler.DependencyAnalysis.NodeFactory>> deferredStaticDependencies) Line 175	C#
 	ILCompiler.DependencyAnalysisFramework.dll!ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer<ILCompiler.DependencyAnalysisFramework.NoLogStrategy<ILCompiler.DependencyAnalysis.NodeFactory>, ILCompiler.DependencyAnalysis.NodeFactory>.ComputeMarkedNodes() Line 315	C#
 	ILCompiler.LLVM.dll!ILCompiler.LLVMCodegenCompilation.CompileInternal(string outputFile, ILCompiler.ObjectDumper dumper) Line 71	C#
 	ILCompiler.Compiler.dll!ILCompiler.Compilation.ILCompiler.ICompilation.Compile(string outputFile, ILCompiler.ObjectDumper dumper) Line 503	C#
 	ilc.dll!ILCompiler.Program.Run(string[] args) Line 796	C#
 	ilc.dll!ILCompiler.Program.Main(string[] args) Line 967	C#
```

Basically for the release input we enter `fgMergeBlockReturn` which doesn't happen with the Debug input.  
```
        if ((block->bbJumpKind == BBJ_RETURN) && ((block->bbFlags & BBF_HAS_JMP) == 0))
        {
            if ((genReturnBB != nullptr) && (genReturnBB != block))
            {
                fgMergeBlockReturn(block);
            }
        }
```
[morphdebug.txt](https://github.com/dotnet/runtimelab/files/7831212/morphdebug.txt)
[morphrelease.txt](https://github.com/dotnet/runtimelab/files/7831213/morphrelease.txt)

Its a little odd as there's not much llvm specific stuff before this phase.  There's no call to `fgSetOptions();` for Wasm/LLVM, it couldn't be that could it?

Thanks
